### PR TITLE
Add allowpreceding support

### PR DESCRIPTION
**Pull request by nick4598**

### CODE REVIEW

**Comment by austeja-bentley** [comment]
I think we could merge this but the build validation seems to fail:
image

**Comment by nick4598** [comment]
> I think we could merge this but the build validation seems to fail: image

I changed it to null comparison instead

**Comment by austeja-bentley** [comment]
One more question - are these changes tested, especially the `findLatestV2CheckpointForChangeset` function (at least manually)?

**Comment by nick4598** [comment]
> One more question - are these changes tested, especially the `findLatestV2CheckpointForChangeset` function (at least manually)?

I haven't tested. Do you know of an iModel, especially one that has a v1 checkpoint without a v2 checkpoint? It would be nice to see us skip over that v1 and still find that preceding v2. Or maybe we could make one somehow, maybe start v2ca and kill it before it finishes in dev.

File: tests/imodels-access-backend-tests/src/BackendiModelsAccess.test.ts [hunk 1]
```diff
@@ -216,6 +218,71 @@ describe("BackendIModelsAccess", () => {
...
+      const queryV2CheckpointParams: CheckpointProps = {
+          accessToken,
+          iTwinId,
+          iModelId: testIModelForRead.id,
+          changeset: {
+            id: nextChangeset.id
+          }
+      };
+
+      // Act
```
Comment (austeja-bentley) [inline tests/imodels-access-backend-tests/src/BackendiModelsAccess.test.ts:237]: nitpick: I think two `act/assert` blocks would group the logic a bit better:
```
// Act
const v2checkpointForExactChangeset: V2CheckpointAccessProps | undefined = await backendIModelsAccess.queryV2Checkpoint(queryV2CheckpointParams);
// Assert
expect(v2checkpointForExactChangeset).to.be.undefined;

// Act
const v2checkpointForChangesetAllowPrecedingParams = {...queryV2CheckpointParams, allowPreceding: true};
const v2checkpointForChangesetAllowPreceding: V2CheckpointAccessProps | undefined = await backendIModelsAccess.queryV2Checkpoint(queryV2CheckpointParamsAllowPreceding);
// Assert
expect(v2checkpointForChangesetAllowPreceding).to.not.be.undefined;
```

**Review by austeja-bentley — COMMENTED** [review]
I added one comment about the test structure and lint for this PR fails on the validation build, after those small fixes I think we can merge this.

File: tests/imodels-access-backend-tests/src/BackendiModelsAccess.test.ts (hunk 1 above)
Comment (nick4598) [reply tests/imodels-access-backend-tests/src/BackendiModelsAccess.test.ts:236]: done

**Review by austeja-bentley — APPROVED** [review]

**Pull request merged**